src/runtime/virtcontainers/hypervisor_config_linux.go

```diff
@@ -44,6 +44,10 @@ func validateHypervisorConfig(conf *HypervisorConfig) error {
 		conf.MemorySize = defaultMemSzMiB
 	}
 
+	if uint64(conf.MemorySize) > conf.DefaultMaxMemorySize {
+		conf.MemorySize = uint32(conf.DefaultMaxMemorySize)
```

Comment:
Shouldn't you issue at least a warning in that case? Why is that not seen as a configuration error? And if anything, shouldn't things happen the other way round, if the objective is to just limit hotplug? The hotplug code compares the current memory size with the host memory size (which is where we take the max memory size into account IIUC). So if the memory size is already larger, should't that be respected?

Again, a bit more documentation in the commit messages about how you want to use this and what the intended effect is would help a lot.

Reply:
Yeah agree, more logging and docs would be worth while let me extend and elaborate on each fo the commits.